src/pkg/task/controller.go

```diff
@@ -0,0 +1,233 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	cjob "github.com/goharbor/harbor/src/common/job"
+	"github.com/goharbor/harbor/src/common/job/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/pkg/task/manager"
+	"github.com/goharbor/harbor/src/pkg/task/model"
+)
+
+var (
+	// Ctl is a global variable for the default task controller implementation
+	Ctl = NewController()
+)
+
+// Controller is an interface for task management. A task is corresponding to a job
+// in Jobservice.
+// All Jobservice job related operations should be covered by the controller to reduce
+// the complexity
+type Controller interface {
+	// Submit creates the task record in database and submits a job to the Jobservice
+	Submit(job *model.Job, option ...*model.Option) (int64, error)
+	Get(id int64) (*model.Task, error)
+	Stop(id int64) error
+	Delete(id int64) error
+	GetLog(id int64) ([]byte, error)
+	HandleHook(id int64, sc *job.StatusChange) error
+	// Calculate the status of a group of tasks specified by the ids.
+	// If the status is a final status, an end time is returned either
+	CalculateTaskGroupStatus(ids []int64) (string, time.Time, error)
```

Comment:
change the argument to group id and return a rich data model with overall status and stats info